R-packages/evalcast/R/get_predictions.R

```diff
@@ -78,6 +77,11 @@ get_predictions_single_date <- function(forecaster,
                                         ...) {
   if (length(geo_values) > 1) geo_values <- list(geo_values)
   forecast_date <- lubridate::ymd(forecast_date)
+  # compute the start_day from the forecast_date, if we need to
+  if (!is.null(signals$start_day) && is.list(signals$start_day) &&
+      is.function(signals$start_day[[1]])) {
+    signals$start_day <- signals$start_day[[1]](forecast_date)
+  }
```

Comment:
I do not believe this accounts for the situation where some signals have a fixed start date and others have a function start date.  Here's an example that gives an error:

```
signals_mix <- tibble(data_source = "jhu-csse",
                      signal = c("deaths_incidence_num", "confirmed_incidence_num"),
                      start_day = list("2020-06-15",
                                       function(forecast_date) forecast_date - 40))
get_predictions(baseline_forecaster,
                name_of_forecaster = "baseline",
                signals = signals_mix,
                forecast_dates = "2020-07-20",
                incidence_period = "epiweek",
                ahead = 3,
                geo_type = "state")
```

Reply:
Ooh I didn't anticipate people would be mixing and matching, though I guess we can allow that.  The documentation (man-roxygen/signals-template.R) doesn't explicitly say it's forbidden, but I guess the way it's written I was thinking people would interpret it, either specify a date or a function.

Reply:
So, I've accounted for a mix of types now.  Your example passes.  Thanks for catching this.